### DISCUSSION TITLE
Are there any examples for using https://www.w3.org/TR/webrtc/#rtcdatachannel

### DISCUSSION THREAD

**Discussion by fire** [elixir-webrtc/ex_webrtc#107] [Q&A] [answered]
I am interested in samples for the rtcdatachannel api for Godot Engine usage.

**Comment by mickel8** [comment, marked as answer]
Hi @fire, support for datachannels is planned but for sure not in 0.3.0. See our roadmap for more ([link](https://github.com/elixir-webrtc/ex_webrtc/issues/28))

**Reply by fire** [reply]
How can I help?

**Reply by LVala** [reply]
Well, the most crucial part of DataChannels is the SCTP - Erlang does implement SCTP but it requires OS support (which, unfortunately, is not guaranteed on systems like Windows and MacOS) and it does all of the IO, which makes it unusable with ICE anyways.

That means the first step to working DataChannels would be to implement Sans-IO-style SCTP library in Elixir, which is a huge undertaking (or wrap existing library in NIF or similar, but it's not that simple and I did not find any suitable library after a quick search). You can try, but I'm not sure if we will be able to help (we would to need to learn the protocol ourselves).
